solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -298,7 +305,8 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     final RTimerTree timer = rb.isDebug() ? req.getRequestTimer() : null;
 
     final ShardHandler shardHandler1 = getAndPrepShardHandler(req, rb); // creates a ShardHandler object only if it's needed
-    
+
+    tagRequestWithRequestId(rb);
```

Comment:
super minor but please add a blank line after this because it's not related to timer.